service_config.md

```diff
@@ -0,0 +1,194 @@
+#  Services configuration / k8s operators
+
+## Goal
+
+Describe an approach that can be adopted across the operators to improve the
+deployment aspect and the way secrets and config files are generated. The
+solution proposed in this document, in the first place, is to give an operator
+the ability to render and inject sensitive snippets within the service config.
+
+### Note:
+
+The document is not supposed to cover the implementation details that might
+converge or diverge according to a given operator, the number of deployed
+services, and other potential requirements that are not the same across the
+board.
+
+
```

Comment:
Does this proposal targets service config for services running on the EDPM nodes or only targeting services running in k8s pods?

Reply:
No, the proposal is for services in the ctlplane: I think we can follow up and expand to the EDPM world as long as we're in agreement on this side.

Reply:
Sure. Let's state that this proposal only targets services running in k8s pods.

Reply:
part of the inital feedback i provided is that we should not try and split this along those lines since there is disagreement on what is the control plane.
just because its on an edpm host does not mean its not part fo the control plane.

with that said if other service s want have different config generation for depm nodes that is there choice but it would be explicitly against one of our early design goals which was to have no config generation in ansible.

for the nova operator anything we implement for openshift will also be used for the edpm nodes
we already excplivly generate service configuration for nova-comptue and libvirt using golang templates.

ovn/neutron however are currently split in how they are configured
hopefully they will see value in converging on this. proposal for there edpm configs too.

Reply:
If it helps settle this sooner then I agree to scope it to services in k8s pods only. While I agree with @SeanMooney that the same logic described here should be applied to services running on EDPM nodes, I prefer settle this with a smaller scope sooner and extending it later, than holding the whole thing up due to the EDPM architectural discussion.

Reply:
+1 I think we're on the same page here.